jax/extend/core/primitives.py

```diff
@@ -0,0 +1,229 @@
+  round_p as round_p,
+  rsqrt_p as rsqrt_p,
+  select_n_p as select_n_p,
+  shift_left_p as shift_left_p,
+  shift_right_arithmetic_p as shift_right_arithmetic_p,
+  shift_right_logical_p as shift_right_logical_p,
+  sign_p as sign_p,
+  sin_p as sin_p,
+  sinh_p as sinh_p,
+  sort_p as sort_p,
+  sqrt_p as sqrt_p,
+  squeeze_p as squeeze_p,
+  sub_p as sub_p,
+  tan_p as tan_p,
+  tanh_p as tanh_p,
+  top_k_p as top_k_p,
+  transpose_p as transpose_p,
+  xor_p as xor_p,
+)
+
+from jax._src.lax.special import (
+  bessel_i0e_p as bessel_i0e_p,
+  bessel_i1e_p as bessel_i1e_p,
+  digamma_p as digamma_p,
+  erfc_p as erfc_p,
+  erf_inv_p as erf_inv_p,
+  erf_p as erf_p,
+  igammac_p as igammac_p,
+  igamma_grad_a_p as igamma_grad_a_p,
+  igamma_p as igamma_p,
+  lgamma_p as lgamma_p,
+  polygamma_p as polygamma_p,
+  random_gamma_grad_p as random_gamma_grad_p,
+  regularized_incomplete_beta_p as regularized_incomplete_beta_p,
+  zeta_p as zeta_p,
+)
+
+from jax._src.lax.slicing import (
+  dynamic_slice_p as dynamic_slice_p,
+  dynamic_update_slice_p as dynamic_update_slice_p,
+  gather_p as gather_p,
+  scatter_add_p as scatter_add_p,
+  scatter_max_p as scatter_max_p,
+  scatter_min_p as scatter_min_p,
+  scatter_mul_p as scatter_mul_p,
+  scatter_p as scatter_p,
+  slice_p as slice_p,
+)
+
+from jax._src.lax.convolution import (
+  conv_general_dilated_p as conv_general_dilated_p,
+)
+
+from jax._src.lax.windowed_reductions import (
+  reduce_window_max_p as reduce_window_max_p,
+  reduce_window_min_p as reduce_window_min_p,
+  reduce_window_p as reduce_window_p,
+  reduce_window_sum_p as reduce_window_sum_p,
+  select_and_gather_add_p as select_and_gather_add_p,
+  select_and_scatter_p as select_and_scatter_p,
+  select_and_scatter_add_p as select_and_scatter_add_p,
+)
+
+from jax._src.lax.control_flow import (
+  cond_p as cond_p,
+  cumlogsumexp_p as cumlogsumexp_p,
+  cummax_p as cummax_p,
+  cummin_p as cummin_p,
+  cumprod_p as cumprod_p,
+  cumsum_p as cumsum_p,
+  linear_solve_p as linear_solve_p,
+  scan_p as scan_p,
+  while_p as while_p,
+)
+
+from jax._src.lax.fft import (
+  fft_p as fft_p,
+)
+
+from jax._src.lax.parallel import (
+  all_gather_p as all_gather_p,
+  all_to_all_p as all_to_all_p,
+  axis_index_p as axis_index_p,
+  pmax_p as pmax_p,
+  pmin_p as pmin_p,
+  ppermute_p as ppermute_p,
+  psum_p as psum_p,
+)
+
+from jax._src.lax.ann import (
+  approx_top_k_p as approx_top_k_p
+)
+
+from jax._src.lax.linalg import (
+  cholesky_p as cholesky_p,
+  eig_p as eig_p,
+  eigh_p as eigh_p,
+  hessenberg_p as hessenberg_p,
+  lu_p as lu_p,
+  householder_product_p as householder_product_p,
+  qr_p as qr_p,
+  svd_p as svd_p,
+  triangular_solve_p as triangular_solve_p,
+  tridiagonal_p as tridiagonal_p,
+  tridiagonal_solve_p as tridiagonal_solve_p,
+  schur_p as schur_p,
+)
+
+from jax._src.pjit import sharding_constraint_p as sharding_constraint_p
+from jax._src.prng import threefry2x32_p as threefry2x32_p
```

Comment:
Should we also have `random_split_p`, `random_fold_in_p`, `random_bits_p`, `random_wrap_p`, `random_unwrap_p`, `random_seed_p`?

Reply:
Good idea. These are not "public" via any other package now, and the same goes for a handful of other primitives. We can add these next, or as needed.